src/GafferOSLModule/GafferOSLModule.cpp

```diff
@@ -140,10 +205,21 @@ BOOST_PYTHON_MODULE( _GafferOSL )
 	def( "oslLibraryVersionPatch", &oslLibraryVersionPatch );
 	def( "oslLibraryVersionCode", &oslLibraryVersionCode );
 
-	IECorePython::RefCountedClass<ShadingEngine, IECore::RefCounted>( "ShadingEngine" )
-		.def( init<const IECore::ObjectVector *>() )
-		.def( "shade", &ShadingEngine::shade, shadeOverloads() )
-	;
+
+	{
+		scope s = IECorePython::RefCountedClass<ShadingEngine, IECore::RefCounted>( "ShadingEngine" )
+			.def( init<const IECore::ObjectVector *>() )
+			.def( "shade", &shadeWrapper, shadeOverloads() )
+		;
+
+		class_<ShadingEngine::Transform>( "Transform" )
+			.def( "__init__", make_constructor( transformConstructor, default_call_policies() ) )
```

Comment:
Is the `make_constructor` stuff actually needed here? Can you not just use `.def( init<const M44f &>() )` and `.def( init<const M44f &, const M44f&>() )`.